### ISSUE TITLE
[NLP] [question/bug?] `train_loss` falling, whereas `val_loss`, `punct_f1`, `capit_f1` all increasing

### ISSUE THREAD

**Issue by itzsimpl** [NVIDIA/NeMo#4575]
@PeganovAnton Training on multiple GPUs I'm noticing that `train_loss` is decreasing, and `f1` scores are increasing, but so is `val_loss`. Is `val_loss` the right metric to be monitored? Would mean `f1` score be better suited?

What does `logits_ndim` in https://github.com/NVIDIA/NeMo/blob/fea3775c00adfacfe0a414dea15544abc96db8dc/nemo/collections/nlp/models/token_classification/punctuation_capitalization_model.py#L126
stand for? Is this initialisation still correct if one changes the number of classes, i.e. includes support for additional punctuation marks? What if the numbers of classes for punctuation and capitalisation differ?  

<img width="348" alt="image" src="https://user-images.githubusercontent.com/4784070/180222056-3f2d6dbc-6123-43a1-a4d1-c49b1651f979.png">
<img width="348" alt="image" src="https://user-images.githubusercontent.com/4784070/180222242-6fde5196-fc70-454e-86c7-fbe3c800f5a0.png">
<img width="348" alt="image" src="https://user-images.githubusercontent.com/4784070/180222345-ee0f65a2-7153-4d8e-b652-2b549e937be0.png">
<img width="348" alt="image" src="https://user-images.githubusercontent.com/4784070/180222409-27f760c0-4d4e-41ef-a58f-0a905d73862f.png">

**Environment details**

`pytorch:22.06-py3` + `nemo:1.10.0` + PR #4553 patch

**Comment by PeganovAnton** [comment]
Hi @itzsimpl ! Sorry for the late response.
1. I also remember occasions when validation loss increases while validation F1 increases. I suggest using F1 as monitored metric.
2. `logits_ndim` [is](https://github.com/NVIDIA/NeMo/blob/fea3775c00adfacfe0a414dea15544abc96db8dc/nemo/collections/common/losses/cross_entropy.py#L48) a dimensionality of the logits tensor. You may increase number of punctuation marks without changing this parameter.
3. You may add any number of new punctuation characters if they follow preceding word and then followed by space, e.g. semicolon, colon, exclamation mark. Though I doubt, that hyphen surrounded by spaces or opening parentheses will work correctly during inference (`add_punctuation_capitalization()` method).
4. You may add more capitalization labels, though you will need to modify `add_punctuation_capitalization()`.
